tests/test_diffraction_objects.py

```diff
@@ -165,99 +167,108 @@ def test_init_invalid_xtype():
-            "d": None,
-            "offset": 2.1,
-        },
-        {"xtype": "tth", "yarray": np.array([4.1, 5.1, 6.1, 7.1, 8.1, 9.1])},
-    ),
-    # UC2: same length x-arrays with exact x-value match
-    (
-        {
-            "xarray": np.array([10, 15, 25, 30, 60, 140]),
-            "yarray": np.array([10, 20, 25, 30, 60, 100]),
-            "xtype": "tth",
-            "wavelength": 2 * np.pi,
-            "target_xarray": np.array([10, 20, 25, 30, 60, 140]),
-            "target_yarray": np.array([2, 3, 4, 5, 6, 7]),
-            "target_xtype": "tth",
-            "target_wavelength": 2 * np.pi,
-            "q": None,
-            "tth": 60,
-            "d": None,
-            "offset": 0,
-        },
-        {"xtype": "tth", "yarray": np.array([1, 2, 2.5, 3, 6, 10])},
-    ),
-    # UC3: same length x-arrays with approximate x-value match
-    (
-        {
-            "xarray": np.array([0.12, 0.24, 0.31, 0.4]),
-            "yarray": np.array([10, 20, 40, 60]),
-            "xtype": "q",
-            "wavelength": 2 * np.pi,
-            "target_xarray": np.array([0.14, 0.24, 0.31, 0.4]),
-            "target_yarray": np.array([1, 3, 4, 5]),
-            "target_xtype": "q",
-            "target_wavelength": 2 * np.pi,
-            "q": 0.1,
-            "tth": None,
-            "d": None,
-            "offset": 0,
-        },
-        {"xtype": "q", "yarray": np.array([1, 2, 4, 6])},
-    ),
-    # UC4: different x-array lengths with approximate x-value match
-    (
-        {
-            "xarray": np.array([10, 25, 30.1, 40.2, 61, 120, 140]),
-            "yarray": np.array([10, 20, 30, 40, 50, 60, 100]),
-            "xtype": "tth",
-            "wavelength": 2 * np.pi,
-            "target_xarray": np.array([20, 25.5, 32, 45, 50, 62, 100, 125, 140]),
-            "target_yarray": np.array([1.1, 2, 3, 3.5, 4, 5, 10, 12, 13]),
-            "target_xtype": "tth",
-            "target_wavelength": 2 * np.pi,
-            "q": None,
-            "tth": 60,
-            "d": None,
-            "offset": 0,
-        },
-        # scaling factor is calculated at index = 4 (tth=61) for self and index = 5 for target (tth=62)
-        {"xtype": "tth", "yarray": np.array([1, 2, 3, 4, 5, 6, 10])},
-    ),
-]
-
-
-@pytest.mark.parametrize("inputs, expected", params_scale_to)
-def test_scale_to(inputs, expected):
-    orig_diff_object = DiffractionObject(
-        xarray=inputs["xarray"], yarray=inputs["yarray"], xtype=inputs["xtype"], wavelength=inputs["wavelength"]
-    )
-    target_diff_object = DiffractionObject(
-        xarray=inputs["target_xarray"],
-        yarray=inputs["target_yarray"],
-        xtype=inputs["target_xtype"],
-        wavelength=inputs["target_wavelength"],
-    )
-    scaled_diff_object = orig_diff_object.scale_to(
-        target_diff_object, q=inputs["q"], tth=inputs["tth"], d=inputs["d"], offset=inputs["offset"]
+@pytest.mark.parametrize(
+    "org_do_args, target_do_args, scale_inputs, expected",
```

Comment:
instead of using `inputs[target_xarray]`, and inputs`[xarray]`, etc. separate args for each DO into variables.